### PR TITLE
test: integration tests

### LINKED ISSUE #614: integration tests for built-ins
**Is your feature request related to a problem? Please describe.**
Currently, since the built-ins only have unit tests we don't know when they break because of some problems interacting with the outside(for example changes in API calls)

**Describe the solution you'd like**
We could have a suite of integration tests that run the built-ins against actual pull requests / issues and check they are working appropriately

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Adds integration tests to reviewpad that creates a PR on a Github repository and runs a set of reviewpad files on the pr.

## Related issue

Closes #614 

## Type of change

Improvements (non-breaking change without functionality)

## How was this tested?
Manual tests

## Checklist

- [x] I have performed a self-review of my code
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: integration_test/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+		createPullRequestInput githubv4.CreatePullRequestInput
+		updatePullRequestInput *githubv4.UpdatePullRequestInput
+		reviewpadFiles         []*engine.ReviewpadFile
+		fileChanges            *githubv4.FileChanges
+		commitMessage          string
+		wantErr                error
+		exitStatus             []engine.ExitStatus
+		cleanup                func(context.Context, *github.GithubClient, string, string, string) error
+	}{
+		"kitchen-sink": {
```
Comment (ferreiratiago) [inline integration_test/integration_test.go:165]: What does `kitchen-sink` means in this context?

**Review by ferreiratiago — CHANGES_REQUESTED** [review]

**Comment by ferreiratiago** [comment]
Can you please create an issue to add this `task integration-test` to the main CI?

File: integration_test/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+	subFile, err := utils.ReadFile("./assets/utils/sub.go")
+	require.Nil(err)
+
+	readmeFile, err := utils.ReadFile("./assets/README.md")
+	require.Nil(err)
+
+	binaryFile, err := utils.ReadFile("./assets/hello-world")
+	require.Nil(err)
+
+	rawMostBuiltInsReviewpadFile, err := utils.ReadFile("./assets/most-built-ins-reviewpad.yml")
```
[the remark this answers is not in the document]
Comment (zolamk) [reply integration_test/integration_test.go:107]: Yeah, makes sense to make it more descriptive

File: integration_test/integration_test.go (hunk 1 above)
Comment (zolamk) [reply integration_test/integration_test.go:165]: It just means to indicate a lot of the built-ins but better to rename it

File: integration_test/assets/utils/add.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2022 Explore.dev Unipessoal Lda. All Rights Reserved.
+// Use of this source code is governed by a license that can be
+// found in the LICENSE file.
+
+// @reviewpad-an: critical
```
[the remark this answers is not in the document]
Comment (zolamk) [reply integration_test/assets/utils/add.go]: Yeah, that's a mistake

**Comment by zolamk** [comment]
> Can you please create an issue to add this `task integration-test` to the main CI?

Sure, of course.

**Review by ferreiratiago — APPROVED** [review]

**Review by shay2025 — APPROVED** [review]
Just some minor sentence changes suggestions 😉 but overall LGTM 🙌

**Pull request merged**